### ISSUE TITLE
using the UWP application to get .PBF extention tiles from a server

### ISSUE THREAD

**Issue by ImCarrot** [ryanlowdermilk/VectorTilesWithXamarinMaps#1]
Hi, Thanks for the sample on vector tiles with the nugget. Just a quick thing, I am using your solution to implement my own vector tiles returned from a server call. The below is my process tile code: 
        `private async Task<bool> ProcessTile(TileData t, Map m, int xOffset, int yOffset)
        {
            int x = t.X + xOffset;
            int y = t.Y + yOffset;

            string key = $"{x}_{y}";

            if (MapStore.XY.ContainsKey(key))
            {
                return false;
            }
            else
            {
                MapStore.XY.Add(key, true);
            }

            Assembly assembly = GetType().GetTypeInfo().Assembly;

            var stringUrl = $"http://MySomeServicePlaceholder/maps/v1/myKey/map_tile/v14_0_z4_ind/{t.Z}/{t.X}/{t.Y}.pbf";

            Uri URL = new Uri(stringUrl);

            using (var client = new HttpClient())
            {
                var requestMessage = new HttpRequestMessage(HttpMethod.Get, URL);
                var responseMessage = await client.SendAsync((requestMessage));
                var responseData = await responseMessage.Content.ReadAsStreamAsync();
                if (responseData != null && responseData.Length > 0)
                {
                    var layerInfos = VectorTileParser.Parse(responseData);

                    if (layerInfos.Count == 0)
                        return false;

                    var fc = layerInfos[0]?.ToGeoJSON(x, y, t.Z);

                    foreach (var geo in fc.Features)
                    {
                        var lng1 = ((GeoJSON.Net.Geometry.GeographicPosition)((GeoJSON.Net.Geometry.Point)geo.Geometry).Coordinates).Longitude;
                        var lat1 = ((GeoJSON.Net.Geometry.GeographicPosition)((GeoJSON.Net.Geometry.Point)geo.Geometry).Coordinates).Latitude;
                        m.Pins.Add(new Pin() { Position = new Position(lat1, lng1), Label = $"{lng1},{lat1}" });
                    }
                }
                else
                    return false;
            }
            return true;
        }
`

The issue is it goes into the catch error section in the method `Map_PropertyChanged` with the error message:
 `Message = "Invalid wire-type; this usually means you have over-written a file without truncating or setting the length; see http://stackoverflow.com/q/2152978/23354"` 

I did try and see the given link but couldn't get it through. Could you please help

**Comment by bertt** [comment]
Probably you need to check the GZip compression, see https://github.com/bertt/mapbox-vector-tile-cs/blob/master/README.md

**Comment by ImCarrot** [comment]
@bertt yes I did read that, I also checked the Tests you wrote, but I am not able find how should I check the tests you wrote, my Visual Studio 2015 Update 3 doesn't find your tests at all so although I am able took inside of your tests I can't seem to get them to run. On another note when you say I need to check the GZip compression does this mean that the data I am getting from my service has an issue with the Gzip compression or is it something do to with the code. I really appreciate you helping me out on this

**Comment by bertt** [comment]
. for running NUnit tests something extra needs to be installed https://marketplace.visualstudio.com/items?itemName=NUnitDevelopers.NUnit3TestAdapter
. GZip compression is just a regular mechanism to compress the resources from the server as much as possible

**Comment by ImCarrot** [comment]
@bertt thank you for the NUnit tests debugging link the issue is now resolved. I don't get the issue about the protoBuffer anymore. It works amazing. But I am still not able to get now that I have a list of `VectorTileLayer` from your samples and the nugget, how do I put the Tile Source of the map to the List of VectorTileLayer that we just decompressed using the nugget and remove the Bing map tile source completly (removing the Bing map tile source I can perform as I've been working with Raster Tiles for quite a while now). I know I am taking up a lot of your time and I am really sorry to bother you this way, I really appreciate your help and  I would be more than happy to extend you the same courtesy anytime.
